action/protocol/staking/vote_bucket.go

```diff
@@ -260,8 +262,29 @@ func delBucket(sm protocol.StateManager, index uint64) error {
 }
 
 func getAllBuckets(sr protocol.StateReader) ([]*VoteBucket, error) {
-	// TODO
-	return nil, nil
+	// bucketKey is prefixed with const bucket = '0', all bucketKey will compare less than []byte{bucket+1}
+	maxKey := []byte{bucket + 1}
+	_, iter, err := sr.States(
+		protocol.NamespaceOption(factory.StakingNameSpace),
+		protocol.FilterOption(func(k, v []byte) bool {
+			return bytes.HasPrefix(k, []byte{bucket})
+		}, bucketKey(0), maxKey))
+	if errors.Cause(err) == state.ErrStateNotExist {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var buckets []*VoteBucket
```

Comment:
you would prealloc slice here. `buckets := make([]*VoteBucket
, 0, iter.Size())`

Reply:
ok